internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/ConfigParser.java

```diff
@@ -0,0 +1,248 @@
+     */
+    public static List<String> parseProductList() throws ConfigParserException {
+        List<String> products;
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        //load a properties file from class path
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.PRODUCT_LIST_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing productList config file : " + e);
+        }
+        products = Arrays.asList(properties.getProperty(JSScannerConstants.PRODUCTS).
+                split(JSScannerConstants.PROPERTIES_FILE_DELIMETER));
+        log.info("[JS_SEC_DAILY_SCAN] Supported products are : " + String.join(",", products));
+        return products;
+    }
+
+
+    /**
+     * This method is responsible to parse properties regarding to upload scan reports into particular github
+     * repository. THe following properties are read from config file.
+     * 1.Username
+     * 2.Password
+     * 3.Repository url where the reports should be uploaded.
+     *
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static ReportUploader parseGitUploaderConfigProperties() throws GitAPIException, ConfigParserException {
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.GIT_CONFIG_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing github credentials : " + e);
+        }
+        ReportUploader reportUploader = new GitUploader(properties.getProperty(JSScannerConstants.USERNAME)
+                .toCharArray(), properties.getProperty(JSScannerConstants.PASSWORD).toCharArray(),
+                properties.getProperty(JSScannerConstants.SECURITY_ARTIFACT_REPO));
+        return reportUploader;
+
+    }
+
+    /**
+     * Parse github credential details (Access-token, Username, Password).
+     *
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static void parseGitAccessToken() throws ConfigParserException {
+
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.GIT_ACCESS_TOKEN_CONFIG_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing github credentials : " + e);
+        }
+        CommonApiInvoker.setGitToken(properties.getProperty(JSScannerConstants.ACCESSTOKEN).toCharArray());
```

Comment:
This should go inside the try block, because this depends on property loading being successful.